src/Stack/Build.hs

```diff
@@ -111,12 +113,14 @@ build setLocalFiles mbuildLk bopts = fixCodePage' $ do
 
     if boptsDryrun bopts
         then printPlan plan
-        else executePlan menv bopts baseConfigOpts locals
-                         globalDumpPkgs
-                         snapshotDumpPkgs
-                         localDumpPkgs
-                         installedMap
-                         plan
+        else do
+            warnIfExecutablesWithSameNameCouldBeOverwritten locals plan
```

Comment:
How about moving this before the `if` so that the warnings are also printed for `--dry-run`?